prov/verbs/src/verbs_cq.c

```diff
@@ -239,18 +239,22 @@ int vrb_poll_cq(struct vrb_cq *cq, struct ibv_wc *wc)
 
 		ctx = (struct vrb_context *) (uintptr_t) wc->wr_id;
 		wc->wr_id = (uintptr_t) ctx->user_ctx;
-		if (ctx->flags & FI_TRANSMIT) {
+		if (ctx->op_ctx == VRB_POST_SQ) {
+			assert(ctx->ep);
+			assert(!slist_empty(&ctx->ep->sq_list));
+			assert(ctx->ep->sq_list.head == &ctx->entry);
```

Comment:
is there an order guarantee while polling? It might not be the first one right?

Reply:
Yes, completion order is guaranteed by the HW.  All operations posted to the send queue will complete in order.

Reply:
in ofi manpage -
For transmit requests, completion ordering depends on the endpoint communication type. For unreliable communication, completion ordering applies to all data transfer requests submitted to an endpoint. For reliable communication, completion ordering only applies to requests that target a single destination endpoint. Completion ordering of requests that target different endpoints over a reliable transport is not defined.

And if user set FI_ORDER_NONE, then "No ordering is defined for completed operations. Requests submitted to the transmit context may complete in any order"

So in verbs provider it internally actually works in "FI_ORDER_STRICT" mode?

Reply:
Correct - verbs HW implements the equivalent of FI_ORDER_STRICT.